app/views/sales/index.html.erb

```diff
@@ -0,0 +1,212 @@
+<% if @is_vendor %>
+<div>
+  <h1><%= @vendor.name %> Sales</h1>
+  <%= link_to "Add a Sale", "/vendors/#{@vendor.id}/sales/new", class: "btn btn-default" %>
+  <div class="container">
+    <h2>All Sales</h2>
+    <table class="table table-bordered">
+      <tr >
+        <th>
+          PRODUCT NAME
+        </th>
+        <th>
+          SALE
+        </th>
+        <th>
+          DATE
+        </th>
+        <th></th>
+      </tr>
+        <% total = 0 %>
+        <% @vendor.products.each do |product| %>
+          <% product.sales.length.times do |i| %>
```

Comment:
Ruby has an `each_with_index` method which will not only give you the individual item (sale in this case) but it will also give you the index. This might be more clear to use rather than using the `i` to retrieve the sale info.